executors/stable/executor.py

```diff
@@ -0,0 +1,154 @@
+import argparse, os, sys, glob
+from io import BytesIO
+import torch
+import numpy as np
+from PIL import Image
+from tqdm import tqdm, trange
+from itertools import islice
+from einops import rearrange
+import time
+from pytorch_lightning import seed_everything
+from torch import autocast
+from contextlib import nullcontext
+from typing import Dict
+from random import randint
+
+from ldm.util import instantiate_from_config
+from ldm.models.diffusion.ddim import DDIMSampler
+from ldm.models.diffusion.plms import PLMSSampler
+
+from jina import Executor, DocumentArray, Document, requests
+from omegaconf import OmegaConf
+
+
+class StableDiffusionConfig():
+    '''
+    Configuration for stable diffusion.
+    '''
+    ckpt = None
+    config = None
+    plms = False
+    seed = randint(0, 2 ** 64)
+    ddim_steps = 100
+    fixed_code = False
+    ddim_eta = 0.0
+    n_iter = 1
+    height = 512
+    width = 512
+    C = 4 # latent channels
+    f = 8 # downsampling factor
+    n_samples = 4
+    scale = 7.5 # unconditional guidance scale
+    precision = 'autocast'
+
+
+def chunk(it, size):
+    it = iter(it)
+    return iter(lambda: tuple(islice(it, size)), ())
+
+
+def load_model_from_config(config, ckpt, verbose=False):
+    print(f"Loading model from {ckpt}")
+    pl_sd = torch.load(ckpt, map_location="cpu")
+    if "global_step" in pl_sd:
+        print(f"Global Step: {pl_sd['global_step']}")
+    sd = pl_sd["state_dict"]
+    model = instantiate_from_config(config.model)
+    m, u = model.load_state_dict(sd, strict=False)
+    if len(m) > 0 and verbose:
+        print("missing keys:")
+        print(m)
+    if len(u) > 0 and verbose:
+        print("unexpected keys:")
+        print(u)
+
+    model.cuda()
+    model.eval()
+    return model
+
+
+class StableDiffusionGenerator(Executor):
+    opt: StableDiffusionConfig = None
```

Comment:
what is the point if I cannot change it?

Reply:
I will default it to an instance and add ways to alter the keys via the `config.yml`.